paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonBaseScan.scala

```diff
@@ -127,7 +127,9 @@ abstract class PaimonBaseScan(
 
   override def estimateStatistics(): Statistics = {
     val stats = PaimonStatistics(this)
-    if (reservedFilters.nonEmpty) {
+    // When statistics doesn't exist, we don't need to perform FilterEstimation on stats,
+    // because in this scenario stats is calculated through splits and partition pruning has been done.
+    if (statistics.isPresent && reservedFilters.nonEmpty) {
```

Comment:
It is better to merge `PaimonStatistics` and here, we'd better maintain only one judgment.